dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPIImpl.java

```diff
@@ -0,0 +1,564 @@
+                    secrets.getServiceKey(), user.getUserId(), host.getIdentifier()));
+        } else {
+            final String internalKey = internalKey(secrets.getServiceKey(), host);
+            secretsStore.saveValue(internalKey, toJsonString(secrets).toCharArray());
+        }
+    }
+
+    @Override
+    public void deleteSecrets(final String serviceKey, final Host host, final User user)
+            throws DotDataException, DotSecurityException {
+        if (userDoesNotHaveAccess(user)) {
+            throw new DotSecurityException(String.format(
+                    "Invalid service delete attempt on `%s` for host `%s` performed by user with id `%s`",
+                    serviceKey, host.getIdentifier(), user.getUserId()));
+        } else {
+            secretsStore.deleteValue(internalKey(serviceKey, host));
+        }
+    }
+
+    private static final String DESCRIPTORS_CACHE_GROUP = "DESCRIPTORS_CACHE_GROUP";
+    private static final String DESCRIPTORS_LIST_KEY = "DESCRIPTORS_LIST_KEY";
+    private static final String DESCRIPTORS_MAPPED_BY_SERVICE_KEY = "DESCRIPTORS_MAPPED_BY_SERVICE_KEY";
+
+    @Override
+    public List<ServiceDescriptor> getServiceDescriptors(User user)
+            throws DotDataException, DotSecurityException {
+        return getServiceDescriptorsMeta(user).stream()
+                .map(ServiceDescriptorMeta::getServiceDescriptor)
+                .collect(Collectors.toList());
+    }
+
+    private List<ServiceDescriptorMeta> getServiceDescriptorsMeta(final User user)
+            throws DotDataException, DotSecurityException {
+        if (userDoesNotHaveAccess(user)) {
+            throw new DotSecurityException(String.format(
+                    "Invalid attempt to get all available service descriptors performed by user with id `%s`.",
+                    user.getUserId()));
+        }
+
+            List<ServiceDescriptorMeta> serviceDescriptors = (List<ServiceDescriptorMeta>) CacheLocator
+                    .getCacheAdministrator().getNoThrow(
+                            DESCRIPTORS_LIST_KEY, DESCRIPTORS_CACHE_GROUP);
+            if (!UtilMethods.isSet(serviceDescriptors)) {
+                synchronized (ServiceIntegrationAPIImpl.class) {
+                    try {
+                        serviceDescriptors = loadServiceDescriptors();
+                    } catch (IOException e) {
+                        Logger.error(ServiceIntegrationAPIImpl.class,
+                                "An error occurred while loading the service descriptor yml files. ",
+                                e);
+                        throw new DotDataException(e);
+                    }
+                    CacheLocator.getCacheAdministrator()
+                            .put(DESCRIPTORS_LIST_KEY, serviceDescriptors, DESCRIPTORS_CACHE_GROUP);
+                }
+            }
+            return serviceDescriptors;
+    }
+
+    private Map<String, ServiceDescriptorMeta> getServiceDescriptorMap(User user)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)